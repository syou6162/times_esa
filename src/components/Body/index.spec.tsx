process.env.REACT_APP_VALID_MAIL_ADDRESSES = 'valid@example.com';

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Body, BodyProps } from '.'

describe('Bodyが正しく表示される', () => {
  it('firebaseの認証が表示されていない状態', () => {
    const props: BodyProps = {
      hasUserLanded: false,
      isSignedIn: false,
      user: {
        email: '',
        displayName: '',
        photoURL: '',
      },
      firebaseAuth: null
    }
    const renderResult = render(
      <Body {...props} />
    );

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
  it('firebaseの認証を通過していない状態', () => {
    const props: BodyProps = {
      hasUserLanded: true,
      isSignedIn: false,
      user: {
        email: '',
        displayName: '',
        photoURL: '',
      },
      firebaseAuth: null
    }
    const renderResult = render(
      <Body {...props} />
    );

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
  it('firebaseの認証が通ったが、invalidなユーザー', () => {
    const props: BodyProps = {
      hasUserLanded: true,
      isSignedIn: true,
      user: {
        email: 'invalid@example.com',
        displayName: 'invalid',
        photoURL: 'invalid',
      },
      firebaseAuth: null
    }
    const renderResult = render(
      <Body {...props} />
    );

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
  // firebaseのモックがまだなので、ひとまずコメントアウト
  // it('firebaseの認証が通ったが、validなユーザー', () => {
  //   const props: BodyProps = {
  //     hasUserLanded: true,
  //     isSignedIn: true,
  //     user: {
  //       email: 'valid@example.com',
  //       displayName: 'valid',
  //       photoURL: 'valid',
  //     },
  //     firebaseAuth: null
  //   }
  //   const renderResult = render(
  //     <Body {...props} />
  //   );

  //   expect(renderResult.asFragment()).toMatchSnapshot();
  // });
})
