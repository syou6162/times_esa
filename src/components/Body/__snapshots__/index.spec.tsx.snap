// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bodyが正しく表示される firebaseの認証が表示されていない状態 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl"
  >
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      #times_esa
    </a>
    : 今日は0個つぶやいたよ
    <form>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root makeStyles-multilineColor-1 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            placeholder="日報のタイトルを記入しましょう"
            required=""
            rows="1"
            style="height: 0px; overflow: hidden;"
          >
            日報
          </textarea>
          <textarea
            aria-hidden="true"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
            tabindex="-1"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-2"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-4"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        aria-expanded="false"
        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon"
        role="combobox"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root makeStyles-multilineColor-7 MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              id="esa_tags_field"
              placeholder="タグを記入しましょう"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-8"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-4"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root makeStyles-multilineColor-9 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            placeholder="ここにつぶやいた内容がesa.ioに追記されていきます"
            rows="6"
            style="height: 0px; overflow: hidden;"
          />
          <textarea
            aria-hidden="true"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
            tabindex="-1"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline makeStyles-notchedOutline-10"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-4"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        title="esa_submit_form_button"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          つぶやく
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
    <hr
      style="border-top: 2px dashed #bbb;"
    />
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall"
        style="margin: 3px; text-transform: none;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Update
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall"
        style="margin: 3px; text-transform: none;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          html
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall"
        style="margin: 3px; text-transform: none;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          text
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall"
        style="margin: 3px; text-transform: none;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          tweet
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div>
        <div
          style="white-space: normal; overflow: hidden; text-align: left; justify-content: left; align-items: left;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Bodyが正しく表示される firebaseの認証が通ったが、invalidなユーザー 1`] = `
<DocumentFragment>
  <div>
    <div>
      ようこそ invalid (
      <img
        alt="invalid"
        height="20px"
        src="invalid"
        title="invalid@example.com"
      />
      invalid@example.com) さん!
    </div>
    <div>
      Error: 有効なメールアドレスではありません。
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Sign-out
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Bodyが正しく表示される firebaseの認証を通過していない状態 1`] = `
<DocumentFragment>
  <div>
    ログインが必要です
  </div>
</DocumentFragment>
`;
